src/traces/scattergl/convert.js

```diff
@@ -112,6 +112,11 @@ function convertTextStyle(trace) {
     var i;
 
     optsOut.text = trace.text;
+    if(Array.isArray(optsOut.text)) {
+        for(i = optsOut.text.length; i < count; i++) {
+            optsOut.text[i] = '';
+        }
+    }
```

Comment:
⚠️ this is mutating the original data array - I don't think we want to do that.

<img width="326" alt="Screen Shot 2019-08-13 at 5 29 39 PM" src="https://user-images.githubusercontent.com/2678795/62978783-11350980-bdf0-11e9-9a36-f3ec682480a2.png">

Reply:
Nice catch @alexcjohnson !

Because this file conflicts with my `texttemplate`, branch I will have to modify it anyway. How should I solve this problem? Can I simply `slice` it or is this a no go for performance?

Reply:
I'd say slice is fine - as long as you only do it when `text.length < count`. I think it's totally reasonable to say "for best performance make sure your arrays are all the same length" and prioritize stability over edge-case performance. The presumably higher-perf alternative would be to track down where the array is being indexed and fall back to a blank string at that point... but looks like that's deep in a regl submodule so that sounds painful 😅

Reply:
Probably not a big deal. But since we're planning on releasing 1.49.3 and this PR got merged, @antoinerg can you make another PR off master to fix and test the `optsOut.text` mutation?

Reply:
.... instead of fixing it in https://github.com/plotly/plotly.js/pull/4071

Reply:
@etpinard Done in https://github.com/plotly/plotly.js/pull/4126